File: frontend/lib/services/route_service.dart
import '../models/route_result.dart';
import 'api_client.dart';

class RouteService {
  Future<RouteResult> calculate({
    required double startLat,
    required double startLng,
    String? shelterId,
  }) async {
    final res = await ApiClient.instance.post(
      '/route/calculate',
      data: {
        'start_lat': startLat,
        'start_lng': startLng,
        'shelter_id': shelterId,
      },
    );
    return RouteResult.fromJson(res.data['data'] as Map<String, dynamic>);
  }
}

File: frontend-rn/src/services/route.service.ts
import { apiClient } from './api-client';
import { routeResultFromJson, type RouteResult } from '@/types/route-result';

export async function calculateRoute(p: {
  startLat: number;
  startLng: number;
  shelterId?: string;
}): Promise<RouteResult> {
  const res = await apiClient.post('/route/calculate', {
    start_lat: p.startLat,
    start_lng: p.startLng,
    shelter_id: p.shelterId ?? null,
  });
  return routeResultFromJson(res.data?.data);
}
